docs/README.md

```diff
@@ -1,63 +1,94 @@
-
-Аналогично реализации MVVM, рекомендованной гуглом, когда у вьюмодели есть несколько Observable полей по типу isLoading: StateFlow<Boolean>, isError: StateFlow<Boolean>.  
-А сама вьюмодель при этом во время работы экрана меняет флаги в этих состояниях.
-
-Чем сложнее становится логика экранов, тем сложнее поддерживать консистентное состояние вью вызывая на ней нужные команды в случае MVP (часто важен даже порядок их вызова) или изменяя несколько Observable состояний.
-
-Отделение логики изменения вью от бизнес-логики позволяет создавать очень сложные экраны. При этом не приходится работать с огромным презентером на несколько тысяч строк, который содержит большое количество логики и чем сложнее эта логика - тем более хрупким становится код в нем.
+mvi-flow уже подерживает *почти* все DSL-трансформации, которые поддерживала предыдущая версия
+Подробнее про DSL можно почитать в [DSL](DSL.md)
+
+## Вычисление диффов при отрисовке состояния
+Если мы используем Android View Framework и просто подпишемся на Observable состояние экрана и будем каждый раз полностью его перерисовывать когда получим новое состояние - на экранах с большим количеством логики и сложными вью может упасть FPS. И чем чаще будет обновляться состояние - тем сильнее будет падение.
+Для этого было сделано просто решение - перерисовывать только те части экрана, которые поменялись или отрисовываются в первый раз.
+На проектах для этого использовалась утилитарная функция `actionIfChanged` или одна из ее перегрузок для разного количества аргументов (обычно больше четырех для одной вью не трубуется, а для особых случаев можно сделать обертку).
+```kotlin
+fun <T : View, R1, R2, R3, R4> T.actionIfChanged(
+    data1: R1?,
+    data2: R2? = null,
+    data3: R3? = null,
+    data4: R4? = null,
+    action: T.(data1: R1?, data2: R2?, data3: R3?, data4: R4?) -> Unit) {
+    val hash = data1?.hashCode() ?: 0
+        .plus(data2?.hashCode() ?: 0)
+        .plus(data3?.hashCode() ?: 0)
+        .plus(data4?.hashCode() ?: 0)
+    if (this.tag != hash) {
+        action(data1, data2, data3, data4)
+        this.tag = hash
+    }
+}
+```
+Но уже очень скоро на всех проектах студии будет использоваться Compose и дополнительные действия не потребуются, так как в Compose подсчет Diff есть из коробки и Compose отлично подходит под работу с единым состоянием.
 
 ## Пример
-
-Реализацию простого экрана с полным набором сущностей и классов можно посмотреть в модуле [sample](../sample)
-
-### Если у вас есть мысли которые помогут развить идеи этой библиотеки
-Telegram: @RuslanSharipov   
-Gmail: ruslan.b.sharipov@gmail.com
+В модуле [sample](../sample) можно посмотреть две реализации с полным набором сущностей.
+- Простая с примерами использования трансформаций и тривиальной логикой экрана
+- С запросом в сеть и обработкой ошибок с помощью ErrorHandler (используется на большинстве проектов студии).
+
+## Roadmap
```

Comment:
I suggest to remove it and to keep such ideas in issues (some of them are added)